### PR TITLE
feat(meetings): show AI summary badge on meeting cards (LFXV2-2208)

### PR DESCRIPTION
**Pull request by audigregorie**
# Summary

Meeting list endpoints return indexed query-service projections where AI summary settings live under nested `zoom_config` rather than the top-level fields the UI reads. This change normalizes those records so `ai_summary_enabled` and `require_ai_summary_approval` are populated from `zoom_config` when absent, letting meeting cards show the AI summary badge consistently.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/shared/src/utils/meeting.utils.ts [hunk 1]
```diff
@@ -621,3 +621,20 @@ export function parseTranscriptVtt(vtt: string | null | undefined): TranscriptCu
...
+  const zoom = meeting.zoom_config;
+  if (!zoom) {
+    return meeting;
+  }
+
+  return {
+    ...meeting,
+    ai_summary_enabled: meeting.ai_summary_enabled ?? zoom.ai_companion_enabled ?? null,
+    require_ai_summary_approval: meeting.require_ai_summary_approval ?? zoom.ai_summary_require_approval ?? null,
+  };
```
[the remark this answers is not in the document]
Comment (audigregorie) [reply packages/shared/src/utils/meeting.utils.ts:673]: Fixed in ff9911a5 — removed trailing `?? null` so absent values stay `undefined` and are omitted from JSON responses. Explicit `null` from `zoom_config` still passes through via `??`.

**Comment by luismoriguerra** [comment]
## Audit: PR #1015 — feat(meetings): show AI summary badge on meeting cards (LFXV2-2208)

**audigregorie** · `feat/lfxv2-2208-ai-summary-badge` → `main` · OPEN · 5 files, +106/−11 · [LFXV2-2208](https://linuxfoundation.atlassian.net/browse/LFXV2-2208)

**Scope:** BFF-side normalization of indexed `v1_meeting` query-service projections so list responses populate top-level `ai_summary_enabled` / `require_ai_summary_approval` from nested `zoom_config`.

**Lanes run:** lint · check-types · format · shared unit tests · secrets · code-standards · architecture · bugbot

---

### 1. Why this change

Meeting cards read top-level `ai_summary_enabled`, but indexed `v1_meeting` projections store AI settings under `zoom_config.ai_companion_enabled`. List/dashboard badges were missing even when AI summary was enabled.

### 2. How it was done

- Added `normalizeIndexedMeetingAiSummary()` in `packages/shared` + `MeetingZoomConfig` interface
- Applied at both indexed list producers: `MeetingService.getMeetings` and `UserService.getUserMeetings`
- Mirrors existing indexed-projection normalization patterns (`normalizeIndexedVote`, past-meeting V1→V2 in the same utils file)
- No UI changes — cards already had `hasAiSummary()` wired

### 3. Is it correct

**Yes.** Both list paths covered; `getMeetingById` correctly left on ITX (top-level fields). Precedence via `??` preserves explicit top-level values including `false`.

**Prior review threads:**

| Review | Status |
|--------|--------|
| Copilot — trailing `?? null` adds JSON null keys | **Fixed** in `ff9911a5` |
| Copilot — unrelated `search-index.json` | **Resolved** — removed from PR |
| CodeRabbit — search-index wording | **N/A** — file removed |
| Copilot re-review at head | No new comments |

**Verification (isolated worktree @ `ff9911a54`):**

| Check | Result |
|-------|--------|
| `yarn workspace @lfx-one/shared test meeting.utils.spec.ts` | ✅ 37 passed |
| `yarn lint` | ✅ |
| `yarn check-types` | ✅ |
| `yarn format:check` | ✅ |
| `yarn build` / `yarn e2e` | ⏭ not run (no UI surface change) |

### 4. Missing / gaps

1. **Non-blocking** — Add util test that `zoom_config` present + absent AI fields → `undefined` (not `null`); guards `ff9911a5` fix
2. **Non-blocking** — Collapse `MeetingZoomConfig` multi-line interface JSDoc to one line (repo convention)

No blocking gaps. Security/architecture/bugbot lanes found no medium+ issues or ADR violations.

### 5. Q&A this PR resolves

- **Why BFF not indexer?** Indexed shape canonically nests AI flags under `zoom_config`; ITX uses top-level fields. BFF bridging at read time matches ADR-0013 drift-tolerant reads.
- **Why `ai_companion_enabled`?** Indexer contract field name; mapped to app `ai_summary_enabled`.
- **Do explicit `false` values get overwritten?** No — `??` only fills absent values.
- **Which views fixed?** Meeting list (`GET /api/meetings`) and dashboard/my-meetings (`GET /api/user/meetings`).

### 6. Business rules & ADRs

- **Business rule:** Top-level AI-summary fields win when present; derive from `zoom_config` only when absent; no-op when `zoom_config` missing.
- **ADR:** None added/superseded. Aligns with ADR-0013(c) and CQRS “different shapes for different jobs.”

### 7. Open questions & confirmations

- **(confirm)** Planned indexer/query-service flattening of top-level `ai_summary_enabled` on `v1_meeting`?
- **(confirm)** Are past-meeting list cards in scope for LFXV2-2208, or upcoming-only? (`v1_past_meeting` uses `zoom_ai_enabled` in indexer.)

---

**Verdict: ✅ Approved**

Merge readiness: **safe to merge** — optional polish on JSDoc + one regression test (inline comments attached).

[LFXV2-2208]: https://linuxfoundation.atlassian.net/browse/LFXV2-2208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: packages/shared/src/interfaces/meeting.interface.ts [hunk 2]
```diff
@@ -131,6 +131,23 @@ export interface MeetingCommittee {
   name?: string;
 }
 
+/**
+ * Zoom-specific configuration as returned on the indexed `v1_meeting` query-service projection.
+ * Note the field names differ from the top-level app contract: the indexer nests AI-summary
+ * enablement here as `ai_companion_enabled` (vs top-level `ai_summary_enabled`) and approval as
+ * `ai_summary_require_approval` (vs top-level `require_ai_summary_approval`).
+ */
```
Comment (luismoriguerra) [inline packages/shared/src/interfaces/meeting.interface.ts]: **Non-blocking (repo convention):** Collapse this multi-line interface JSDoc to a single line — per-field one-liners below are fine, but the interface-level block should be one self-contained `/** … */` per the UI pre-merge checklist.

Suggested:
`/** Indexed v1_meeting zoom_config; AI flags use ai_companion_enabled / ai_summary_require_approval vs top-level ai_summary_enabled / require_ai_summary_approval. */`

File: packages/shared/src/utils/meeting.utils.spec.ts [hunk 3]
```diff
@@ -136,6 +142,49 @@ describe('resolveOccurrenceRecurrence', () => {
...
+  it('derives require_ai_summary_approval from zoom_config with the same precedence', () => {
+    const fromZoom = { zoom_config: { ai_summary_require_approval: true } } as Meeting;
+    expect(normalizeIndexedMeetingAiSummary(fromZoom).require_ai_summary_approval).toBe(true);
+
+    const topLevelWins = {
+      require_ai_summary_approval: false,
+      zoom_config: { ai_summary_require_approval: true },
+    } as Meeting;
+    expect(normalizeIndexedMeetingAiSummary(topLevelWins).require_ai_summary_approval).toBe(false);
+  });
```
Comment (luismoriguerra) [inline packages/shared/src/utils/meeting.utils.spec.ts:185]: **Non-blocking (regression coverage):** `ff9911a54` removed trailing `?? null` so absent values stay `undefined` — worth a test guarding that behavior when `zoom_config` is present but neither nested nor top-level AI fields are set:

```typescript
it('leaves ai_summary fields undefined when neither layer provides a value', () => {
  const meeting = { zoom_config: { meeting_id: '123' } } as Meeting;
  const result = normalizeIndexedMeetingAiSummary(meeting);
  expect(result.ai_summary_enabled).toBeUndefined();
  expect(result.require_ai_summary_approval).toBeUndefined();
});
```

(Copilot's `?? null` thread is resolved — this is test coverage for the fix, not a re-open of that finding.)

**Review by luismoriguerra — COMMENTED** [review]
Inline notes from the audit (full report posted as a top-level comment). Verdict: **Approved** — no blockers.

**Review by luismoriguerra — DISMISSED** [review]

File: packages/shared/src/interfaces/meeting.interface.ts (hunk 2 above)
Comment (audigregorie) [reply packages/shared/src/interfaces/meeting.interface.ts]: Done in 8c1ef1c3 — collapsed to a single-line interface JSDoc.

File: packages/shared/src/utils/meeting.utils.spec.ts (hunk 3 above)
Comment (audigregorie) [reply packages/shared/src/utils/meeting.utils.spec.ts:185]: Done in 8c1ef1c3 — added regression test for undefined AI fields when zoom_config is present but neither layer provides a value.

**Review by MRashad26 — COMMENTED** [review]
## Review — PR #1015 `feat/lfxv2-2208-ai-summary-badge`

**Verdict: PASS**

### Overview
Adds `normalizeIndexedMeetingAiSummary()` to `@lfx-one/shared/utils` to bridge the field-name mismatch between query-service indexed projections (`zoom_config.ai_companion_enabled`) and the top-level fields the UI reads (`ai_summary_enabled`). Applied in both the meeting list path (`meeting.service.ts`) and the user meetings path (`user.service.ts`). Comes with comprehensive unit tests.

### Secrets check
Clean — no credentials or API keys.

### Code standards — all clear
- ✅ `MeetingZoomConfig` interface in `packages/shared/src/interfaces/meeting.interface.ts`
- ✅ `zoom_config?: MeetingZoomConfig | null` added to `Meeting` in same file
- ✅ `normalizeIndexedMeetingAiSummary()` in `packages/shared/src/utils/meeting.utils.ts`
- ✅ No template changes — no Angular template violations possible
- ✅ No `## Test plan` section in PR body

### Code correctness — no issues

**Precedence logic** — `??` (nullish coalescing) correctly lets `false` at the top level win over a `true` in `zoom_config`. Top-level `undefined`/`null` falls through to `zoom_config`. Tests cover all four cases ✅

**Identity shortcut** — returns the input reference unchanged when `zoom_config` is absent, avoiding unnecessary object allocation ✅

**Generic constraint** — `<T extends Pick<Meeting, ...>>` keeps the return type narrowed to `T` rather than widening to `Meeting`, so callers retain their specific type ✅

**`user.service.ts` mutation safety** — the RSVP enrichment loop mutates `meeting.my_rsvp` on each element of `normalizedMeetings`. When `zoom_config` is present, `normalizeIndexedMeetingAiSummary` returns a new spread object so the mutation is isolated. When `zoom_config` is absent the same reference is returned — but that's identical to the pre-PR behavior, not a regression introduced here ✅

**Test coverage** — 7 cases: both flags from `zoom_config`, top-level wins in both directions, identity return, combined flags, undefined-when-neither-layer-present. Complete ✅

No issues found.

**Review by MRashad26 — APPROVED** [review]
